In [ ]:
import os
import joblib

print("Model file exists:", os.path.exists(model_path))

loaded_ada_model = joblib.load(model_path)

print("Loaded model type:", type(loaded_ada_model))
print("Loaded model:", loaded_ada_model)

In [9]:
import joblib

model_path = "../models/adaboost_baseline.joblib"

joblib.dump(ada_model, model_path)

print("AdaBoost model saved successfully.")
print("Model path:", model_path)

AdaBoost model saved successfully.
Model path: ../models/adaboost_baseline.joblib


In [8]:
ada_results = {
    "Model": "AdaBoost",
    "Accuracy": ada_accuracy,
    "Precision": ada_precision,
    "Recall": ada_recall,
    "F1 Score": ada_f1
}

print(ada_results)

{'Model': 'AdaBoost', 'Accuracy': 0.9999427794864459, 'Precision': 0.9999427944018742, 'Recall': 0.9999427794864459, 'F1 Score': 0.9999427793706248}


In [7]:
from sklearn.metrics import (
    accuracy_score,
    precision_score,
    recall_score,
    f1_score,
    classification_report
)

ada_accuracy = accuracy_score(y_test, y_pred_ada)

ada_precision = precision_score(
    y_test,
    y_pred_ada,
    average="weighted",
    zero_division=0
)

ada_recall = recall_score(
    y_test,
    y_pred_ada,
    average="weighted",
    zero_division=0
)

ada_f1 = f1_score(
    y_test,
    y_pred_ada,
    average="weighted",
    zero_division=0
)

print("AdaBoost Baseline Metrics")
print("--------------------------------")
print("Accuracy :", ada_accuracy)
print("Precision:", ada_precision)
print("Recall   :", ada_recall)
print("F1 Score :", ada_f1)

print("\nClassification Report:")
print(classification_report(
    y_test,
    y_pred_ada,
    zero_division=0
))

AdaBoost Baseline Metrics
--------------------------------
Accuracy : 0.9999427794864459
Precision: 0.9999427944018742
Recall   : 0.9999427794864459
F1 Score : 0.9999427793706248

Classification Report:
                precision    recall  f1-score   support

        Benign       1.00      1.00      1.00    133525
FTP-BruteForce       1.00      1.00      1.00     38672
SSH-Bruteforce       1.00      1.00      1.00     37518

      accuracy                           1.00    209715
     macro avg       1.00      1.00      1.00    209715
  weighted avg       1.00      1.00      1.00    209715



In [6]:
print("Generating AdaBoost predictions...")

y_pred_ada = ada_model.predict(X_test_clean)

print("Prediction completed successfully.")
print("Number of predictions:", len(y_pred_ada))

Generating AdaBoost predictions...
Prediction completed successfully.
Number of predictions: 209715


In [5]:
print("Starting AdaBoost training...")

ada_model.fit(X_train_clean, y_train)

print("AdaBoost training completed successfully.")

Starting AdaBoost training...
AdaBoost training completed successfully.


In [4]:
from sklearn.ensemble import AdaBoostClassifier

ada_model = AdaBoostClassifier(
    random_state=42
)

print("AdaBoost model created successfully.")
print(ada_model)

AdaBoost model created successfully.
AdaBoostClassifier(random_state=42)


In [3]:
# Copy train/test inputs before cleaning
X_train_clean = X_train.copy()
X_test_clean = X_test.copy()

# Convert infinite values to NaN
X_train_clean = X_train_clean.replace([np.inf, -np.inf], np.nan)
X_test_clean = X_test_clean.replace([np.inf, -np.inf], np.nan)

# Calculate medians using training data only
train_medians = X_train_clean.median()

# Fill missing values using training medians
X_train_clean = X_train_clean.fillna(train_medians)
X_test_clean = X_test_clean.fillna(train_medians)

# Verify cleaned data
print("X_train_clean NaN:", X_train_clean.isna().sum().sum())
print("X_test_clean NaN :", X_test_clean.isna().sum().sum())

print("X_train_clean Inf:", np.isinf(X_train_clean).sum().sum())
print("X_test_clean Inf :", np.isinf(X_test_clean).sum().sum())

print("\nTraining input is finite:",
      np.isfinite(X_train_clean.to_numpy()).all())

print("Testing input is finite:",
      np.isfinite(X_test_clean.to_numpy()).all())

X_train_clean NaN: 0
X_test_clean NaN : 0
X_train_clean Inf: 0
X_test_clean Inf : 0

Training input is finite: True
Testing input is finite: True


In [2]:
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split

df = pd.read_csv("../cic.csv")

X = df.drop(columns=["Label"])
y = df["Label"]

X_numeric = X.select_dtypes(include="number")

X_train, X_test, y_train, y_test = train_test_split(
    X_numeric,
    y,
    test_size=0.2,
    random_state=42,
    stratify=y
)

print("Training features:", X_train.shape)
print("Testing features :", X_test.shape)
print("Training target  :", y_train.shape)
print("Testing target   :", y_test.shape)

Training features: (838860, 78)
Testing features : (209715, 78)
Training target  : (838860,)
Testing target   : (209715,)


In [1]:
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split

df = pd.read_csv("../cic.csv")

X = df.drop(columns=["Label"])
y = df["Label"]

X_numeric = X.select_dtypes(include="number")

X_train, X_test, y_train, y_test = train_test_split(
    X_numeric,
    y,
    test_size=0.2,
    random_state=42,
    stratify=y
)

print("Training features:", X_train.shape)
print("Testing features :", X_test.shape)
print("Training target  :", y_train.shape)
print("Testing target   :", y_test.shape)

Training features: (838860, 78)
Testing features : (209715, 78)
Training target  : (838860,)
Testing target   : (209715,)


# Day 13 – AdaBoost Training

## Objective
Train the AdaBoost baseline model using the common CIC-IDS2018 training input.

## Model
AdaBoost Classifier

## Training Input
- Training features: 78 numeric features
- Training rows: 838,860
- Target: `Label`
- Number of target classes: 3

## Evaluation
The trained model will be evaluated on the test set using:
- Accuracy
- Weighted Precision
- Weighted Recall
- Weighted F1 Score
- Classification Report

## Day 13 Scope
- Load the dataset.
- Create the common train/test split.
- Handle invalid numeric values using training-data-derived medians.
- Train the AdaBoost model.
- Generate test predictions.
- Calculate baseline evaluation metrics.
- Save the trained model.

Only the AdaBoost model will be trained on Day 13.